pkg/util/pki/csr.go

```diff
@@ -303,6 +325,22 @@ func GenerateTemplate(crt *v1.Certificate) (*x509.Certificate, error) {
 		return nil, err
 	}
 
+	var extraNames []pkix.AttributeTypeAndValue = []pkix.AttributeTypeAndValue{}
+	if crt.Spec.RawSubject != "" {
+		rdnSequence, err := ParseSubjectStringToRdnSequence(crt.Spec.RawSubject)
+		if err != nil {
+			return nil, err
+		}
+
+		// todo: Appending extra names like this satisfies our requirement but has different semantics compared to the literal subject
```

Comment:
Should we rename `RawSubject` to something like `extraNames` to make it very clear that these will be additional individual items in the sequence and not within a sequence? So that adding + in the subject will get lost and everything flattened. This allows us to keep this behaviour whch is "good enough for us". However, I think  having the ability to set the verbatim subject as well would be richer.

Reply:
I'm not too sure I understand what you mean here - could you provide an example?

Reply:
The RDNSequence is a list of lists of key value types. The problem as described in the original issue s that there is a technical difference between:
`List(List(CN=foo), List(OU=A, OU=B), List(C=Bar)))` and `List(List(CN=foo), Lst(OU=A), List(OU=B), List(C=Bar))`. Cert-manager currently outputs the former. 

With this, suggested, implementation we take a literal string, say, `CN=foo, OU=A, OU=B, C=Bar` and add all the elements to the extra names. In this example that is fine because they are all a individual items anyway. However, by saying that it is a literal subject, someone might actually put in `CN=Foo, OU=A+OU=B, C=Bar` which would be `List(List(OU=A, OU=B))`. This implementation would flatten that and yield `CN=Foo, OU=A, OU=B, C=Bar`, so without the + again.